### ISSUE TITLE
Thirst

### ISSUE THREAD

**Issue by herecydev** [smudge202/clean-living#2]
We need the ability to track our level of thirst.
Thirst should diminish over time.
Thirst should be replenished by drinking.
The player should be adversely affected by dehydration
The player should have positive effects when thirst is quenched

**Comment by smudge202** [comment]
This is hopefully possible given the engine features in PR #8

**Comment by mattridgway** [comment]
As per comments on #9 - one of the acceptance criteria on here is thirst should be replenished by drinking - actions should be given to the engine (eg. Do(drinkwater) ) the thirst component then picks this up and can handle it? 

The engine has a message bus for incoming actions (like drink, move, pick up), and each component picks these up and reacts accordingly - such as inventory gains item, movement gets slower because of weight.

thoughts?

**Comment by smudge202** [comment]
I agree, though not entirely sure how that's going to look yet.  I guess I'll soon find out :)

Moving this onto a new branch based on engine-time (#8).  Going to merge that PR so that @herecydev  can work on Hunger #1  at the same time.  Dual stream madness!

**Comment by mattridgway** [comment]
sounds good! will come have a gander later

**Comment by smudge202** [comment]
I think this should be blocked pending #11
